api/src/main/java/com/redhat/ipaas/api/v1/rest/util/PaginationFilter.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.api.v1.rest.util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+public class PaginationFilter<T> implements Function<List<T>, List<T>> {
+
+    private final int startIndex;
+    private final int endIndex;
+
+    @Override
+    public List<T> apply(List<T> list) {
+        if (startIndex >= list.size()) {
```

Comment:
What happens when `startIndex == list.size()` ? IMO the `.sublist()` below will break for such a start index.